### ISSUE TITLE
Questions regarding performance on GSM8kaug

### ISSUE THREAD

**Issue by mfahes** [facebookresearch/coconut#45]
Hi, 

I have few questions regarding CoCoNut applied to GSM8k.

1- What is the accuracy after each stage of training?

2- Is the number of generated steps at inference (on the test set) consistent with "ground truth total steps - number of removed steps"? For instance, taking the following example from the test set:

Q: " John drives for 3 hours at a speed of 60 mph and then turns around because he realizes he forgot something very important at home..."
CoT: [
  "<<3*60=180>>",
  "<<4-2=2>>",
  "<<30*.5=15>>",
  "<<2-.5=1.5>>",
  "<<80*1.5=120>>",
  "<<120+15=135>>",
  "<<180-135=45>>"
]
A: 45

Is the model necessarily generating 6 steps at stage 1, 5 steps at stage 2 and 0 steps in at stage 3? (by steps I mean explicit ones, after latent reasoning operations)

3- Related to the question above, how do you guarantee that at the final stage the model doesn't output any explicit CoT and that it generates directly the answer after latent thought phase?

4- Did you compute the test accuracy at iteration 0 of stage 1? I'm interested in knowing how much decrease in accuracy is solely due to change in the input sequence.

5- Are \<bot\> and \<eot\> tokens necessary? Did you test training all stages without them?

6- For experiments with Llama, what is the accuracy with SFT-CoT? Do you also have stage-wise accuracy scores?

Thank you.

**Comment by Ber666** [comment]
Hi,

Thanks for the questions!

1. On the validation set, the accuracies were: 0.40 (after stage 0), 0.35 (after stage 1), 0.32 (after stage 2), 0.33 (after stage 3), and 0.35 for the fully latent model after training to convergence.

2. Not necessarily. We do not impose constraints on decoding that force the model to generate a specific number of explicit steps. The model decides when to stop and can produce eos token autonomously.

3. In the final stage, the training targets always contain only the final answer. As a result, the trained model learns the consistent behavior to produce the answer directly.

4. It's around 0.30.

5. Good question. I did not run an ablation removing them across all stages, but my expectation is that they would not have a large effect.

6. Sorry I don't have this data available now. The internship ended and I wasn't able to retain a full log.
